File: java/src/main/java/com/sopra/TPFinal/model/Expertise.java
package com.sopra.TPFinal.model;

import javax.persistence.Column;
import javax.persistence.EmbeddedId;
import javax.persistence.Entity;
import javax.persistence.Table;
import javax.persistence.Version;

@Entity
@Table(name = "expertise")
public class Expertise {
	@EmbeddedId
	@Column(name = "key")
	private ExpertisePK key;
	@Version
	private int version;

	public Expertise() {
	}

	public ExpertisePK getExpertisePK() {
		return key;
	}

	public void setExpertisePK(ExpertisePK expertisePK) {
		this.key = expertisePK;
	}

	public int getVersion() {
		return version;
	}

	public void setVersion(int version) {
		this.version = version;
	}

}

File: angular/src/app/model/expertise.ts
import {ExpertisePK} from './expertise-pk';

export class Expertise {
  constructor(
    private _key?: ExpertisePK) {
  }

  get key(): number {
    return this._key;
  }

  set key(value: number) {
    this._key = value;
  }

}
